feat: integrate Markprompt chat with custom trigger and global styles

letiescanciano opened the pull request:
Closes https://github.com/airbytehq/airbyte-internal-issues/issues/11873
## What

This PR integrates the Markprompt chat assistant into the Docusaurus site, providing users with an AI-powered chat experience accessible from every page. It introduces a custom chat trigger and ensures the chat dialog overlays correctly above all content.

## How

- Added @markprompt/react and @markprompt/css as dependencies.
- Created a MarkpromptChat component and a custom ChatTrigger button, including associated CSS modules for styling.
- Swizzled `<SearchBar />` to add `<ChatTrigger />`on the Navbar
- Updated CSS to ensure the Markprompt chat sheet dialog appears above all other UI elements by increasing its z-index and applying custom styles.


## User Impact
<!--
* What is the end result perceived by the user?
* If there are negative side effects, please list them. 
-->

## Can this PR be safely reverted and rolled back?
<!--
* If unsure, leave it blank.
-->
- [x] YES 💚
- [ ] NO ❌

letiescanciano committed:
feat: integrate Markprompt chat with custom trigger and global styles

- Add @markprompt/react and @markprompt/css dependencies
- Implement MarkpromptChat and ChatTrigger components with custom styling
- Add Octavia images for chat avatar
- Update global and component CSS for Markprompt sheet z-index and appearance

letiescanciano commented:
@ian-at-airbyte I still need access to Markprompt to get some test keys and test locally that everything works as it should, but UI should be ready to check :)

ian-at-airbyte commented:
Thanks Leti! It looks like it's working.

Can we remove the "Powered by MarkPrompt" banner in the chat? I doubt there's a setting in the MarkPrompt config for this, but if necessary we could use display:none to hide it with CSS.

Is there any possibility of having the "Ask AI" button in the navbar instead of floating in the corner? I can foresee some circumstances where it will cover other UI elements.

letiescanciano committed:
move trigger to Navbar

letiescanciano commented on docusaurus/docusaurus.config.js:
```diff
@@ -1,9 +1,6 @@
 // @ts-check
 // Note: type annotations allow type checking and IDEs autocompletion
 import "dotenv/config.js";
-const yaml = require("js-yaml");
-const fs = require("node:fs");
-const path = require("node:path");
```
unused imports

letiescanciano commented:
@ian-at-airbyte I'm sorry I didn't properly read where/how you wanted the chat button.
I've added it to the right side of the navbar as requested. I explored different implementations and swizzling `<SearchBar />` was the one I felt it made more sense and would be easier to maintain.
Let me know if this aligns with what you have in mind.

ian-at-airbyte reviewed (commented):
This is awesome! Yes, exactly what I was thinking. I think using the search bar component was probably the best choice.

Just a few small CSS nitpicks:

- I think the z-index for the dark background when Octavia is active needs to be higher. It should also cover the navbar, since that's disabled while the chat is open.

- In light mode, it looks like the color of the cards on the home page has changed. Not sure if this was intentional - I'm fine either way I think.

- In dark mode, the chat has a couple of white strips at the top and bottom. Perhaps we can override those with our colors.

- There's something weird happening with alignment in the navbar. Not sure if this is easily fixable. I've attached an image of what I mean.

<img width="396" alt="image" src="https://github.com/user-attachments/assets/7da56c5f-629a-438e-a950-e605c64506ff" />

Otherwise, this seems perfect.

letiescanciano committed:
fix button size

letiescanciano commented:
thanks @ian-at-airbyte 
I've addressed your comments:
 (2) - totally unintentional  and (4) - let me know if it looks better now, since Marprompt component wraps the button I'm finding some issues, but can iterate a little bit more if you feel is not right yet.

For 1 and 3, I'm not sure what you mean. 
It seems that the chat is covering the navbar on my end.
![Screenshot 2025-05-22 at 10 07 43](https://github.com/user-attachments/assets/30c4f66f-6c4f-4fe4-9d5b-f16c80d0562a)

And for 3, 
do you mean these lines? Maybe I can also adjust the background, which seems too black and not aligned with our dark background

![Screenshot 2025-05-22 at 10 07 43](https://github.com/user-attachments/assets/c37e0b51-ed88-48f5-ad6d-0d4e94d08a1a)

letiescanciano committed:
style dialog

letiescanciano committed:
add tracking

letiescanciano committed:
increase overlay z-index

letiescanciano committed:
Merge branch 'master' into feat/integrate-markprompt

letiescanciano commented:
@ian-at-airbyte great eye! I didn't notice :) I've increased the overlay z-index to go above the Navbar

ian-at-airbyte reviewed (approved):
No issues from my end. I think we're good to go.
